checker/src/org/checkerframework/checker/units/UnitsAnnotatedTypeFactory.java

```diff
@@ -326,17 +467,376 @@ public Void visitCompoundAssignment(CompoundAssignmentTree node, AnnotatedTypeMi
+
+                    // the Boolean returned is false if the expression contains
+                    // any operator or argument for which it cannot evaluate
+
+                    // if successfully, set exponent to the evaluated value,
+                    // otherwise set to 0.0
+                    exp = evalValue.second ? evalValue.first : 0.0;
+                } else {
+                    // if the second argument is not a mathematical expression
+                    // nor a numerical literal, then do nothing
+                    return null;
+                }
+
+                // see what is the exponent
+                if (exp == 1.0) {
+                    // return the unit of the first parameter
+                    type.replaceAnnotations(lht.getAnnotations());
+                } else if (exp == 2.0) {
+                    // detect the unit of the first parameter
+                    if (UnitsRelationsTools.hasSpecificUnit(lht, m)) {
+                        type.replaceAnnotation(m2);
+                    } else if (UnitsRelationsTools.hasSpecificUnit(lht, mm)) {
+                        type.replaceAnnotation(mm2);
+                    } else if (UnitsRelationsTools.hasSpecificUnit(lht, km)) {
+                        type.replaceAnnotation(km2);
+                    } else {
+                        // if the base unit is something other than m, mm, or km, then return unknown
+                        type.replaceAnnotation(TOP);
+                    }
+                } else if (exp == 3.0) {
+                    // detect the unit of the first parameter
+                    if (UnitsRelationsTools.hasSpecificUnit(lht, m)) {
+                        type.replaceAnnotation(m3);
+                    } else if (UnitsRelationsTools.hasSpecificUnit(lht, mm)) {
+                        type.replaceAnnotation(mm3);
+                    } else if (UnitsRelationsTools.hasSpecificUnit(lht, km)) {
+                        type.replaceAnnotation(km3);
+                    } else {
+                        // if the base unit is something other than m, mm, or km, then return unknown
+                        type.replaceAnnotation(TOP);
+                    }
+                } else if (exp == 0.5) {
+                    // taking the power of 0.5 is the same as taking the square
+                    // root
+                    return processSquareRoot(lht, type);
+                } else if (exp == (1.0d / 3)) {
+                    // taking the power of 1/3 is the same as taking the cube
+                    // root
+                    return processCubeRoot(lht, type);
+                } else {
+                    // return UnknownUnits
+                    type.replaceAnnotation(TOP);
+                }
+
+            } else {
+                // for any other math library method call, return super's result
+                return super.visitMethodInvocation(node, type);
+            }
+
+            return null;
```

Comment:
This is reached when `return UnknownUnits` is reached two elses earlier. Why doesn't this also call the super implementation?